server/app/controllers/admin/AdminProgramBlockQuestionsController.java

```diff
@@ -144,4 +145,34 @@ public Result setOptional(
     return redirect(
         controllers.admin.routes.AdminProgramBlocksController.edit(programId, blockDefinitionId));
   }
+
+  /** POST endpoint for changing position of a question in its block. */
+  @Secure(authorizers = Labels.CIVIFORM_ADMIN)
+  public Result move(
+      Request request, long programId, long blockDefinitionId, long questionDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    DynamicForm requestData = formFactory.form().bindFromRequest(request);
+    int newPosition = Integer.parseInt(requestData.get("position"));
```

Comment:
Is sharing constants through views standard practice? I do want to share constant, just not sure what is the best place for it.

Reply:
Unfortunately sharing constants isn't widely done throughout the codebase, but I think we should try and move towards this model. IMO, the controller already has a dependency on the view, so adding the constant there doesn't introduce any new dependencies.

Another way this can be achieved is by creating a distinct form object with getters/setters and putting the constants there. In this case though, it'd be a single-field form, so likely not really worth adding.

Reply:
Done. Added constant to view. IN this specific that controller doesn't depend on that view. But I think adding this dependency isn't bad as generally they are related and controllers can depend on view.